browser/ui/views/download/brave_download_item_view.cc

```diff
@@ -0,0 +1,220 @@
+    if (needs_repaint)
+      SchedulePaint();
+  }
+
+  // Update tooltip.
+  base::string16 new_tip = brave_model_.GetTooltipText(font_list_, kTooltipMaxWidth);
+  if (new_tip != tooltip_text_) {
+    tooltip_text_ = new_tip;
+    TooltipTextChanged();
+  }
+}
+
+// Positioning routines.
+
+int BraveDownloadItemView::GetYForFilenameText() const {
+  int text_height = font_list_.GetHeight();
+  if (!origin_url_text_.empty())
+    text_height +=
+        kBraveVerticalTextPadding + origin_url_font_list_.GetHeight();
+  if (!status_text_.empty())
+    text_height += kBraveVerticalTextPadding + status_font_list_.GetHeight();
+  return (height() - text_height) / 2;
+}
+
+int BraveDownloadItemView::GetYForOriginURLText() const {
+  int y = GetYForFilenameText();
+  y += (font_list_.GetHeight() + kBraveVerticalTextPadding);
+  return y;
+}
+
+int BraveDownloadItemView::GetYForStatusText() const {
+  int y = GetYForOriginURLText();
+  if (!origin_url_text_.empty())
+    y += (origin_url_font_list_.GetHeight() + kBraveVerticalTextPadding);
+  return y;
+}
+
+// Drawing routines.
+
+void BraveDownloadItemView::DrawStatusText(gfx::Canvas* canvas) {
+  if (status_text_.empty() || IsShowingWarningDialog())
+    return;
+
+  int mirrored_x = GetMirroredXWithWidthInView(
+      kStartPadding + DownloadShelf::kProgressIndicatorSize +
+          kProgressTextPadding,
+      kTextWidth);
+  canvas->DrawStringRect(status_text_, status_font_list_, GetDimmedTextColor(),
+                         gfx::Rect(mirrored_x, GetYForStatusText(), kTextWidth,
+                                   status_font_list_.GetHeight()));
+}
+
+void BraveDownloadItemView::DrawOriginURL(gfx::Canvas* canvas) {
+  if (origin_url_text_.empty() || IsShowingWarningDialog())
+    return;
+
+  int x = kStartPadding + DownloadShelf::kProgressIndicatorSize +
+          kProgressTextPadding;
+  int text_width = kTextWidth;
+ 
+  if (!is_origin_url_secure_) {
+    DrawLockIcon(canvas);
+    int dx = origin_url_font_list_.GetBaseline() + kOriginURLIconRightPadding;
+    x += dx;
+    text_width -= dx;
+  }
+
+  base::string16 originURL = gfx::ElideText(
+      origin_url_text_, origin_url_font_list_, text_width, gfx::ELIDE_TAIL);
+  int mirrored_x = GetMirroredXWithWidthInView(x, text_width);
+
+  canvas->DrawStringRect(
+      originURL, origin_url_font_list_, GetDimmedTextColor(),
+      gfx::Rect(mirrored_x, GetYForOriginURLText(), text_width,
+                origin_url_font_list_.GetHeight()));
+  }
+
+void BraveDownloadItemView::DrawLockIcon(gfx::Canvas* canvas) {
+  if (origin_url_text_.empty() || IsShowingWarningDialog())
+    return;
+
+  int mirrored_x = GetMirroredXWithWidthInView(
+      kStartPadding + DownloadShelf::kProgressIndicatorSize +
+          kProgressTextPadding,
+      kTextWidth);
+
+  // Get lock icon of the font baseline height size.
+  int dy = origin_url_font_list_.GetBaseline();
+  int y = GetYForOriginURLText() + (origin_url_font_list_.GetHeight() - dy) / 2;
```

Comment:
@petemill, I was trying to align the bottom of the icon with the bottom of the text like it is done in Muon:
![downloaditemmuon](https://user-images.githubusercontent.com/41635752/46814924-c79b0e80-cd47-11e8-9b40-2a991ade644d.png)

However, there, the icon height might be a bit smaller so that it's only slightly taller than the initial 'h'.

The new icon seems to be aligning with the bottom of the font:
![downloaditemicon](https://user-images.githubusercontent.com/41635752/46815127-442ded00-cd48-11e8-9723-aa90ac25b915.png)

But, here's the placement of the icon of size dy in a rectangle of the same size drawn at the same coordinates as the icon:
![downloaditemiconrect](https://user-images.githubusercontent.com/41635752/46815261-8ce5a600-cd48-11e8-913a-cc5c59d5f64c.png)

Reply:
@petemill It would probably be easier if the lock icon was anchored at the bottom of the icon image. Here is what it would look like if I shrunk the icon by 1 off of the font's baseline height:

![adjustments](https://user-images.githubusercontent.com/41635752/46819382-932c5000-cd51-11e8-89c9-ea8682379808.jpg)

Reply:
That does look better. I think the aim normally here is to center align the graphic against the portion of the text between the baseline and the "x-height", i.e. center aligned to the text ignoring the ascenders and descenders.

Reply:
@petemill  @mkarolin def vertically align against x-height since it's the weighted center of the text. there is a possibility this could even use one extra pixel push down. tough to judge at the scaled examples

Reply:
Updated the code to use decremented by 1 height.

![adjusted-1](https://user-images.githubusercontent.com/41635752/46823635-b4466e00-cd5c-11e8-871b-bd5c273b6e95.jpg)